StreamChatClient_v3/APIClient/RequestEncoder.swift

```diff
@@ -0,0 +1,194 @@
+        var request: URLRequest
+        
+        do {
+            // Prepare the URL
+            var url = try encodeRequestURL(for: endpoint, queryItems: endpoint.queryItems)
+            url = try url.appendingQueryItems(["api_key": apiKey.apiKeyString])
+            
+            // Create a request
+            request = URLRequest(url: url)
+            request.httpMethod = endpoint.method.rawValue
+            
+            // Encode endpoint-specific query items
+            if let queryItems = endpoint.queryItems {
+                try encodeJSONToQueryItems(request: &request, data: queryItems)
+            }
+            
+            try encodeRequestBody(request: &request, endpoint: endpoint)
+            
+        } catch {
+            completion(.failure(error))
+            return
+        }
+        
+        if endpoint.requiresConnectionId {
+            assert(connectionIdProviderDelegate != nil,
+                   "The endpoind requiers `connectionId` but `connectionIdProviderDelegate` is not set.")
+            
+            connectionIdProviderDelegate?.provideConnectionId { (connectionId) in
+                guard let connectionId = connectionId else {
+                    completion(.failure(ClientError.MissingConnectionId("Failed to get `connectionId`, request can't be created.")))
+                    return
+                }
+                
+                do {
+                    request.url = try request.url?.appendingQueryItems(["connection_id": connectionId])
+                } catch {
+                    completion(.failure(error))
+                    return
+                }
+                
+                completion(.success(request))
+            }
+            
+        } else {
+            completion(.success(request))
+        }
+    }
+    
+    init(baseURL: URL, apiKey: APIKey) {
+        self.baseURL = baseURL
+        self.apiKey = apiKey
+    }
+    
+    // MARK: - Private
+    
+    private func encodeRequestURL<T: Decodable>(for endpoint: Endpoint<T>, queryItems: Encodable?) throws -> URL {
+        var urlComponents = URLComponents()
+        urlComponents.scheme = baseURL.scheme
+        urlComponents.host = baseURL.host
+        urlComponents.path = baseURL.path
+        
+        guard var url = urlComponents.url else {
+            throw ClientError.InvalidURL("URL can't be created using components: \(urlComponents)")
+        }
+        
+        url = url.appendingPathComponent(endpoint.path)
+        return url
+    }
+    
+    private func encodeRequestBody<T: Decodable>(request: inout URLRequest, endpoint: Endpoint<T>) throws {
+        switch endpoint.method {
+        case .get:
+            try encodeGETRequestBody(request: &request, endpoint: endpoint)
+        case .post:
+            try encodePOSTRequestBody(request: &request, endpoint: endpoint)
+        case .delete:
```

Comment:
I think it should be encoded as `GET`.

Reply:
Cool, I wasn't sure.

Reply:
Fixed in 50097e4bc814f08ae8c863a8d2209e39b8555fdb